lib/ansible/cli/galaxy.py

```diff
@@ -957,9 +975,13 @@ def execute_remove(self):
 
     def execute_list(self):
         """
```

Comment:
This PR increases the complexity of `execute_list` a lot:

devel:
```
galaxy.py:958:4: GalaxyCLI.execute_list (complexity 13)
```

This PR:
```
galaxy.py:976:4: GalaxyCLI.execute_list (complexity 31)
```

I'd recommend breaking this method up into a method for each "type". As it is, I feel like it makes it difficult to review.

Reply:
I agree, it's really bloated now. Now that I have it working, I'll simplify it.

Reply:
I have replace `execute_list()` with two separate methods and move the nested methods to private class methods. Things are looking much better now.
```
galaxy.py:1035:4: GalaxyCLI.execute_role_list (complexity 9)
galaxy.py:1077:4: GalaxyCLI.execute_collection_list (complexity 9)
```